### ISSUE TITLE
Error when using CarafeFPN neck replace FPN

### ISSUE THREAD

**Issue by CuongNN218** [open-mmlab/mmdetection#4696]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
I try to replace FPN by FPN Carafe on VFNet config and got this error. I looking for solution in issues but it doesn't help. My training process work well with FPN Necks
The error i in the image below:
![image](https://user-images.githubusercontent.com/26113323/109463242-3e5f1080-7a97-11eb-91ae-b913f442105f.png)

**Reproduction**

1. What command or script did you run?

python3 tools/train.py configs/vfnet/vfnet_r50_carafe_1x_coco.py

2. Did you make any modifications on the code or config? Did you understand what you have modified?
Here is my config file
`model = dict(
    type='VFNet',
    pretrained='torchvision://resnet50',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch'),
        neck=dict(
            type='FPN_CARAFE',
            in_channels=[256, 512, 1024, 2048],
            out_channels=256,
            num_outs=5,
            start_level=1,
            end_level=-1,
            norm_cfg=None,
            act_cfg=None,
            order=('conv', 'norm', 'act'),
            upsample_cfg=dict(
                type='carafe',
                up_kernel=5,
                up_group=1,
                encoder_kernel=3,
                encoder_dilation=1,
                compressed_channels=64)),
    bbox_head=dict(
        type='VFNetHead',
        num_classes=14,
        in_channels=256,
        stacked_convs=3,
        feat_channels=256,
        strides=[8, 16, 32, 64, 128],
        regress_ranges=((-1, 64), (64, 128), (128, 256), (256, 512),
                        (512, 1e10)),
        anchor_generator=dict(
            type='AnchorGenerator',
            ratios=[1.0],
            octave_base_scale=8,
            scales_per_octave=1,
            center_offset=0.0,
            strides=[8, 16, 32, 64, 128]),
        center_sampling=False,
        dcn_on_last_conv=False,
        use_atss=True,
        use_vfl=True,
        loss_cls=dict(
            type='VarifocalLoss',
            use_sigmoid=True,
            alpha=0.75,
            gamma=2.0,
            iou_weighted=True,
            loss_weight=1.0),
        loss_bbox=dict(type='GIoULoss', loss_weight=1.5),
        loss_bbox_refine=dict(type='GIoULoss', loss_weight=2.0)))`
3. What dataset did you use?
I use my dataset custom which have 14 classes and irherrited from COCO dataset
**Environment**
mmdetection 2.8.0

**Comment by jshilong** [comment]
Is this the first iteration?

**Comment by CuongNN218** [comment]
No, It ramdomly appeared when I'm training. when I print the pred and weight out. I saw that pred and weights are nan and pred has (n, 4) shape and weight has (n) shape

**Comment by jshilong** [comment]
You can add `grad_clip` or set more warm-up iterations to avoid nan.

**Comment by jshilong** [comment]
Feel free to reopen the issue if there is any question.
